win32ss/user/ntuser/menu.c

```diff
@@ -4191,10 +4200,21 @@ static INT FASTCALL MENU_TrackMenu(PMENU pmenu, UINT wFlags, INT x, INT y,
                     {
                         executedMenuId = MENU_ButtonUp( &mt, pmMouse, wFlags);
 
-                    /* End the loop if executedMenuId is an item ID */
-                    /* or if the job was done (executedMenuId = 0). */
-                        fRemove = (executedMenuId != -1);
-                        fInsideMenuLoop = !fRemove;
+                        pWnd = ValidateHwndNoErr(mt.TopMenu->hWnd);
+                        /* Exit system menu if system icon is clicked a second time */
+                        if (!firstClick && GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU)
```

Comment:
@Doug-Lyons Could you check, in this case here (and the same one above for button-down) where this `GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU` check returns TRUE (there is equality), whether there is also equality between `mt.TopMenu` and `pmMouse` ?

Reply:
At both locations, testing for the equality produces ```TRUE```. Thoughts? Thanks.

Reply:
Suggestion for a second test: I wonder whether, when the `GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU` is FALSE on the contrary, whether `mt.TopMenu` and `pmMouse` are different in that case.

----

My underlying motivation for these tests suggestions, is because checking whether the TopMenu is equal to pmMouse or not, seems to be what's prefered in our existing code for doing something that looks similar to what the HTSYSMENU test in this PR was for:
https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/ntuser/menu.c;hb=b8abfa76ca99b99d48575ee78e91e273eed1a12b#l3622

Reply:
These two comparisons do not always track as you can see here:
```
(win32ss/user/ntuser/menu.c:4206) err: mt.TopMenu == pmMouse is 1
(win32ss/user/ntuser/menu.c:4207) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 1

(win32ss/user/ntuser/menu.c:4183) err: mt.TopMenu == pmMouse is 1
(win32ss/user/ntuser/menu.c:4184) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 1

(win32ss/user/ntuser/menu.c:4206) err: mt.TopMenu == pmMouse is 1
(win32ss/user/ntuser/menu.c:4207) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 1

(win32ss/user/ntuser/menu.c:4206) err: mt.TopMenu == pmMouse is 1
(win32ss/user/ntuser/menu.c:4207) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 0

(win32ss/user/ntuser/menu.c:4183) err: mt.TopMenu == pmMouse is 0
(win32ss/user/ntuser/menu.c:4184) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 0

(win32ss/user/ntuser/menu.c:4206) err: mt.TopMenu == pmMouse is 0
(win32ss/user/ntuser/menu.c:4207) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 
```
But they are very close. Maybe close enough to use the other condition here?

Reply:
This is what I would have expected. They are "the same" (so to speak), except for this one case:
```
(win32ss/user/ntuser/menu.c:4206) err: mt.TopMenu == pmMouse is 1
(win32ss/user/ntuser/menu.c:4207) err: GetNCHitEx(pWnd, mt.Pt) == HTSYSMENU is 0
```
I wonder where the mouse was (and other conditions) when this happened.

Reply:
@HBelusca, If you just click slowly and release on any of the normal pull-down menus (File | Edit | etc.) you will see the disagreements. Here is my patch with the extra debug info that you can test yourself. Thanks.

[fix_bars_I_Kill_Bugs-CORE-15945.patch](https://github.com/reactos/reactos/files/15211334/fix_bars_I_Kill_Bugs-CORE-15945.patch)